app/scenes/CollectionDelete.js

```diff
@@ -1,62 +1,63 @@
 // @flow
-import { observable } from "mobx";
-import { inject, observer } from "mobx-react";
+import { observer } from "mobx-react";
 import * as React from "react";
-import { withRouter, type RouterHistory } from "react-router-dom";
-import CollectionsStore from "stores/CollectionsStore";
-import UiStore from "stores/UiStore";
+import { useTranslation, Trans } from "react-i18next";
+import { useHistory } from "react-router-dom";
 import Collection from "models/Collection";
 import Button from "components/Button";
 import Flex from "components/Flex";
 import HelpText from "components/HelpText";
+import useStores from "hooks/useStores";
 import { homeUrl } from "utils/routeHelpers";
 
 type Props = {
-  history: RouterHistory,
   collection: Collection,
-  collections: CollectionsStore,
-  ui: UiStore,
   onSubmit: () => void,
 };
 
-@observer
-class CollectionDelete extends React.Component<Props> {
-  @observable isDeleting: boolean;
+function CollectionDelete({ collection, onSubmit }: Props) {
+  const [isDeleting, setIsDeleting] = React.useState();
+  const { ui } = useStores();
+  const history = useHistory();
+  const { t } = useTranslation();
 
-  handleSubmit = async (ev: SyntheticEvent<>) => {
-    ev.preventDefault();
-    this.isDeleting = true;
+  const handleSubmit = React.useCallback(
+    async (ev: SyntheticEvent<>) => {
+      ev.preventDefault();
+      setIsDeleting(true);
 
-    try {
-      await this.props.collection.delete();
-      this.props.history.push(homeUrl());
-      this.props.onSubmit();
-    } catch (err) {
-      this.props.ui.showToast(err.message, { type: "error" });
-    } finally {
-      this.isDeleting = false;
-    }
-  };
+      try {
+        await collection.delete();
+        history.push(homeUrl());
+        onSubmit();
+      } catch (err) {
+        ui.showToast(err.message, { type: "error" });
+      } finally {
+        setIsDeleting(false);
+      }
+    },
+    [ui, onSubmit, collection, history]
```

Comment:
`onSubmit` is an inline function in the parent component. So, I believe a new onSubmit will be created for every render? If so, then using React.useCallback is useless here.

Reply:
Yea, it may not be doing much for now but I think we should aim to `useCallback` everywhere in the parent too in future changes.